cilium/cmd/config.go

```diff
@@ -70,20 +76,55 @@ func configDaemon(cmd *cobra.Command, opts []string) {
 		}
 	} else if len(opts) == 0 {
 		if command.OutputJSON() {
-			if err := command.PrintOutput(cfgStatus.Realized.Options); err != nil {
-				os.Exit(1)
+			if listReadOnlyConfigurations {
+				if err := command.PrintOutput(cfgStatus.DaemonConfigurationMap); err != nil {
+					Fatalf("Error while printing configmap: %v", err)
+				}
+				return
+			} else {
+				if err := command.PrintOutput(cfgStatus.Realized); err != nil {
+					os.Exit(1)
+				}
+				return
 			}
-			return
 		}
-		dumpConfig(cfgStatus.Immutable)
-		dumpConfig(cfgStatus.Realized.Options)
-		fmt.Printf("%-24s %s\n", "k8s-configuration", cfgStatus.K8sConfiguration)
-		fmt.Printf("%-24s %s\n", "k8s-endpoint", cfgStatus.K8sEndpoint)
-		fmt.Printf("%-24s %s\n", "PolicyEnforcement", cfgStatus.Realized.PolicyEnforcement)
-		if cfgStatus.NodeMonitor != nil {
-			fmt.Printf("%-24s %d\n", "MonitorNumPages", cfgStatus.NodeMonitor.Npages)
+		if listReadOnlyConfigurations {
+			fmt.Print("### Read-Only Configurations ###\n\n")
+			keys := make([]string, 0, len(cfgStatus.DaemonConfigurationMap))
+			for k := range cfgStatus.DaemonConfigurationMap {
+				keys = append(keys, k)
+			}
+			sort.Strings(keys)
+			for _, k := range keys {
+				v := cfgStatus.DaemonConfigurationMap[k]
+				if reflect.ValueOf(v).Kind() == reflect.Map {
+					mapString := make(map[string]string)
+					m, ok := v.(map[string]interface{})
+					if ok {
+						fmt.Println(k)
+						for key, value := range m {
+							mapString[key] = fmt.Sprintf("%v", value)
+						}
+						dumpConfig(mapString, true)
+						continue
+					} else {
+						fmt.Fprintf(os.Stderr, "Error: cannot cast daemon config map to map[string]interface{}\n")
+					}
+				}
+				fmt.Printf("%-34s: %v\n", k, v)
+			}
+			fmt.Printf("%-34s: %s\n", "k8s-configuration", cfgStatus.K8sConfiguration)
+			fmt.Printf("%-34s: %s\n", "k8s-endpoint", cfgStatus.K8sEndpoint)
+			if cfgStatus.NodeMonitor != nil {
+				fmt.Printf("%-34s: %d\n", "MonitorNumPages", cfgStatus.NodeMonitor.Npages)
+			}
+			dumpConfig(cfgStatus.Immutable, false)
+		} else {
+			fmt.Print("##### Read Write Configurations #####\n")
+			dumpConfig(cfgStatus.Realized.Options, false)
+			fmt.Printf("%-34s: %s\n", "PolicyEnforcement", cfgStatus.Realized.PolicyEnforcement)
+			return
 		}
-		return
```

Comment:
nit: It's slightly odd to remove the return from after the if/else check and only add the `return` inside one of the `if` / `else` statements, maybe the return should just remain at the end like it was before?

There's a similar thing happening on lines 83, 88. Both sides of the conditional statement execute `return`, so there could just be one `return` statement after the `if`/`else` rather than two.

Functionally there's no difference though and this code is not likely to break in future because the `return`s are in unusual places so I'm OK to just let this be.